### PR TITLE
ensure stable mode dependencies are production-ready, reduce dependencies, and move experimental dependencies into a new package (breaking changes)

### PR DESCRIPTION
**Pull request by bdemann**
## Contributor

- [x] Code has been declaratized
- [x] All new functions have JSDoc/Rustdoc comments
- [x] Error handling beautiful (no unwraps or expects etc)
- [x] Code tested thoroughly
- [x] PR title:
    - [x] `feat:` prefix used if functionality should be included in the `Features` section of the release notes
    - [x] Not sentence cased
    - [x] Described well for release notes
    - [x] Indicates breaking changes with suffix "(breaking changes)"
- [x] Related issues have been linked and all tasks have been completed or made into separate issues
- [x] New documentation enumerated in [the release issue](https://github.com/demergent-labs/azle/issues/2053)
- [x] All breaking changes
    - [x] Described below in the "Breaking Changes" section
    - [x] Migration path described
- [x] Review is requested when ready

## Reviewer

- [x] Code has been declaratized
- [x] All new functions have JSDoc/Rustdoc comments
- [x] Error handling beautiful (no unwraps or expects etc)
- [x] Code tested thoroughly
- [x] PR title:
    - [x] `feat:` prefix used if functionality should be included in the `Features` section of the release notes
    - [x] Not sentence cased
    - [x] Described well for release notes
    - [x] Indicates breaking changes with suffix "(breaking changes)"
- [x] Related issues have been linked and all tasks have been completed or made into separate issues
- [x] New documentation enumerated in [the release issue](https://github.com/demergent-labs/azle/issues/2053)
- [x] All breaking changes
    - [x] Described below in the "Breaking Changes" section
    - [x] Migration path described

## Breaking Changes

- Many of Azle's dependencies have been removed or moved into Azle's `devDependencies` via the  `azle-experimental-deps` npm package
    - If your project relied on any of those dependencies you will have to install them directly or through installing `azle-experimental-deps`
- Projects that use Azle's experimental mode will have to install `azle-experimental-deps`

To install `azle-experimental-deps` run:
```
npm install https://github.com/demergent-labs/azle-experimental-deps#34cf2e34958e03cae7bf63f77f86ae682eef8a7c
```

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -147,7 +147,7 @@ jobs:
         with:
             directories: ${{ matrix.test_group.directories }}
             exclude-dirs: ${{ needs.get-exclude-dirs.outputs.exclude-dirs }}
-            link-azle: ${{ needs.workflow-config.outputs.link-azle == 'true' }}
+            azle-source-mode: 'pack'
```
Comment (bdemann) [inline .github/workflows/test.yml:150]: We need to have a design discussion about when to use pack and when to use link

File: src/exports/_internal/test/deep_equal.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import { jsonStringify } from '#lib/json';
+
+// TODO move this out of the export folder
```
Comment (bdemann) [inline src/exports/_internal/test/deep_equal.ts:3]: resolve this TODO and double check this file

File: src/stable/test/jest_link.ts [hunk 3]
```diff
@@ -1,16 +1,26 @@
 import { execSync } from 'child_process';
-import { join, resolve } from 'path';
+import { join } from 'path';
+
+import { AZLE_ROOT } from '#build/utils/global_paths';
```
Comment (bdemann) [inline src/stable/test/jest_link.ts:4]: Is this a problem? Should we be getting the azle root another way instead of import.meta?

Comment (bdemann) [reply src/stable/test/jest_link.ts:4]: The old way I'm guessing is going to work just fine

File: src/stable/build/index.ts [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+        return;
+    }
+
+    if (command === 'generate') {
+        await handleGenerateCommand();
+
+        return;
+    }
+
+    // @ts-ignore stable extension still runs this command TODO prevent that
```
Comment (bdemann) [inline src/stable/build/index.ts:77]: Let's have a design discussion about this

Comment (bdemann) [reply src/stable/build/index.ts:77]: Do we want to at least make sure that we check that they aren't expecting files to be uploaded? Much in the same way as the original

File: package.json [hunk 5]
```diff
@@ -48,47 +50,22 @@
...
         "@eslint/js": "^9.24.0",
-        "@types/deep-equal": "^1.0.4",
-        "@types/fs-extra": "11.0.4",
-        "@types/pako": "^2.0.3",
         "@types/semver": "^7.7.0",
         "@types/uuid": "^10.0.0",
         "@typescript-eslint/eslint-plugin": "^8.29.1",
         "@typescript-eslint/parser": "^8.29.1",
-        "deep-equal": "^2.2.3",
+        "azle-experimental-deps": "github:demergent-labs/azle-experimental-deps",
```
Comment (bdemann) [inline package.json:68]: I'm guessing you want this locked at a specific hash?

File: src/stable/build/index.ts (hunk 4 above)
Comment (bdemann) [reply src/stable/build/index.ts:77]: Let's make a post install command instead

File: package.json (hunk 5 above)
Comment (bdemann) [reply package.json]: I went a head and did it just in case

File: package.json [hunk 6]
```diff
@@ -48,47 +50,22 @@
...
-        "@types/pako": "^2.0.3",
         "@types/semver": "^7.7.0",
         "@types/uuid": "^10.0.0",
         "@typescript-eslint/eslint-plugin": "^8.30.1",
         "@typescript-eslint/parser": "^8.30.1",
-        "deep-equal": "^2.2.3",
+        "azle-experimental-deps": "github:demergent-labs/azle-experimental-deps#a8ab3214c9a3599ea540d0f429fe0d4b778417cb",
         "eslint": "^9.24.0",
         "eslint-config-prettier": "^10.1.2",
         "eslint-plugin-simple-import-sort": "^12.1.1",
```
Comment (lastmjs) [inline package.json:71]: How is `jest` being used by developer that use `npx azle new`? It seems like when they run `npm test` there should be a failure to find `jest`. Please figure out why this is. Is response to this, we might want to make a `azle-test-deps`.

File: package.json [hunk 7]
```diff
@@ -48,47 +50,22 @@
...
-        "@types/fs-extra": "11.0.4",
-        "@types/pako": "^2.0.3",
         "@types/semver": "^7.7.0",
         "@types/uuid": "^10.0.0",
         "@typescript-eslint/eslint-plugin": "^8.30.1",
         "@typescript-eslint/parser": "^8.30.1",
-        "deep-equal": "^2.2.3",
+        "azle-experimental-deps": "github:demergent-labs/azle-experimental-deps#a8ab3214c9a3599ea540d0f429fe0d4b778417cb",
         "eslint": "^9.24.0",
         "eslint-config-prettier": "^10.1.2",
```
Comment (lastmjs) [inline package.json:70]: Remove the `version` from the `azle-experimental-deps` package.json

File: README.md [hunk 8]
```diff
@@ -35,6 +35,12 @@ This mode is intended for developers who are willing to accept the risk of using
 
 > NOTE: Keep clearly in mind that the experimental mode fundamentally changes the Azle Wasm binary. It is not guaranteed to be secure or stable in API changes or runtime behavior. If you enable the experimental mode, even if you only use APIs from the stable mode, you are accepting a higher risk of bugs, errors, crashes, security exploits, breaking API changes, etc.
```
Comment (lastmjs) [inline README.md:37]: Explore automatically doing this npm install, but with the correct hash pulled from azle's package.json, and install it into the developer's package.json

Comment (lastmjs) [reply README.md:37]: Remove this from the README

File: src/build.ts [hunk 9]
```diff
@@ -53,274 +32,57 @@ process.on('unhandledRejection', (reason: any) => {
...
+        | Command
+        | ExperimentalCommand
+        | undefined;
 
-    if (experimental === false) {
-        checkForExperimentalDfxJsonFields(canisterConfig);
+    if (
+        command === 'build' ||
+        command === 'post-install' ||
+        command === 'upload-assets'
```
Comment (lastmjs) [inline src/build.ts:71]: We need to somehow tell the user that upload-assets can only run in experimental mode. Right now it will just fail in an error saying the upload-assets command does not exist unless there is an experimental flag or environment variable or the flag in dfx.json...

File: src/stable/test/jest_link.ts [hunk 10]
```diff
@@ -1,16 +1,26 @@
 import { execSync } from 'child_process';
-import { join, resolve } from 'path';
+import { join } from 'path';
+
```
Comment (lastmjs) [inline src/stable/test/jest_link.ts:3]: Let's get rid of all of this machinery across the entire PR, because the `npm` tests during the release process should do essentially the same thing.

File: package.json (hunk 7 above)
Comment (bdemann) [reply package.json:70]: Done

File: src/experimental/build/index.ts [hunk 11]
```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env -S npx tsx
+
+import { IOType } from 'child_process';
+import { join } from 'path';
+
+import { build as stableBuild } from '#build/index';
+import { runCommand as runDevTemplateCommand } from '#commands/dev/template';
+import { runCommand as runNewCommand } from '#commands/new';
+import { runCommand as runExperimentalBuildCommand } from '#experimental/commands/build/index';
```
[the remark this answers is not in the document]
Comment (bdemann) [reply src/experimental/build/index.ts:9]: Done

File: src/stable/test/jest_link.ts (hunk 10 above)
Comment (bdemann) [reply src/stable/test/jest_link.ts:3]: I'm going to do this last of all because I need it to test everything about this pr

File: src/build.ts (hunk 9 above)
Comment (bdemann) [reply src/build.ts:71]: done

File: src/stable/test/jest_link.ts (hunk 10 above)
Comment (bdemann) [reply src/stable/test/jest_link.ts:3]: I also added some pack hack to assertAzleExperimentalDeps that will need to be cleaned up

Comment (bdemann) [reply src/stable/test/jest_link.ts:3]: I am wondering if it would be wise to at least leave the plumbing around in the workflow. At least while I am testing this it is very handy to be able to turn it on when I need.

Comment (bdemann) [reply src/stable/test/jest_link.ts:3]: Though by that same argument maybe we should leave it here too? This isn't exactly production code that is going in people's canisters. We don't have to run the pack mode regularly, but maybe it would be nice to have it around just in case?

File: README.md (hunk 8 above)
Comment (bdemann) [reply README.md:37]: Done, but I'm not really happy with it, we should explore it a little more

File: src/stable/test/deep_equal.ts [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+ * @internal
+ *
+ * Checks if two objects are deeply equal.
+ *
+ * @param obj1 The first object to compare.
+ * @param obj2 The second object to compare.
+ * @returns `true` if the objects are deeply equal, `false` otherwise.
+ *
+ * @remarks
+ * It first serializes both objects using our custom jsonStringify function to handle potential problems with Principal
```
Comment (lastmjs) [inline src/stable/test/deep_equal.ts:13]: Comments should explain that not only Principal is the problem

File: src/stable/test/deep_equal.ts [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+ * then parses them back and performs a recursive comparison.
+ */
+export function deepEqual(obj1: any, obj2: any): boolean {
+    // We have historically had issues with various deepEqual implementations most tracing back to subtle differences in versions etc that have caused false negatives.
+    // Our jsonStringify takes out a lot of the possible oddities by serializing them into a more standard format.
+    // So until we convert this to jest and use it's various equality functions, this should be good enough.
```
Comment (lastmjs) [inline src/stable/test/deep_equal.ts:19]: its

File: src/stable/test/deep_equal.ts [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+        return obj1Json === obj2Json;
+    }
+
+    // Check if they have the same constructor (handles arrays vs objects)
+    if (obj1Json.constructor !== obj2Json.constructor) {
+        return false;
+    }
+
+    // Handle arrays specifically
+    if (Array.isArray(obj1Json)) {
```
Comment (lastmjs) [inline src/stable/test/deep_equal.ts:41]: Check that obj2Json is an array before proceeding

File: src/stable/test/deep_equal.ts [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+    // Get keys and compare lengths
+    const keys1 = Object.keys(obj1Json);
+    const keys2 = Object.keys(obj2Json);
+    if (keys1.length !== keys2.length) {
+        return false;
+    }
+
+    // Recursively compare all properties
+    for (const key of keys1) {
+        if (!keys2.includes(key) || !deepEqual(obj1Json[key], obj2Json[key])) {
```
Comment (lastmjs) [inline src/stable/test/deep_equal.ts:62]: Explicit comparisons everywhere

File: src/stable/test/deep_equal.ts [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+
+    // Get keys and compare lengths
+    const keys1 = Object.keys(obj1Json);
+    const keys2 = Object.keys(obj2Json);
+    if (keys1.length !== keys2.length) {
+        return false;
+    }
+
+    // Recursively compare all properties
+    for (const key of keys1) {
```
Comment (lastmjs) [inline src/stable/test/deep_equal.ts:61]: Review with LLMs to ask about any missing edge cases.

File: src/stable/test/deep_equal.ts [hunk 17]
```diff
@@ -0,0 +1,68 @@
...
+        if (!keys2.includes(key) || !deepEqual(obj1Json[key], obj2Json[key])) {
+            return false;
+        }
+    }
+
+    return true;
+}
```
Comment (lastmjs) [inline src/stable/test/deep_equal.ts:68]: Please make more declarative by putting each type of check into its own function

**Review by lastmjs — CHANGES_REQUESTED** [review]

File: examples/experimental/demo/basic_bitcoin/package.json [hunk 18]
```diff
@@ -8,6 +8,7 @@
     },
     "dependencies": {
         "azle": "0.31.0",
+        "azle-experimental-deps": "github:demergent-labs/azle-experimental-deps",
```
[the remark this answers is not in the document]
Comment (bdemann) [reply examples/experimental/demo/basic_bitcoin/package.json:11]: done

File: src/build.ts [hunk 19]
```diff
@@ -1,29 +1,10 @@
...
-    experimentalMessageDfxJson
-} from '#experimental/utils/experimental_message';
-import { Command as ExperimentalCommand } from '#experimental/utils/types';
 import { getCanisterConfig } from '#utils/get_canister_config';
-import { AZLE_ROOT } from '#utils/global_paths';
-import { CanisterConfig, Command, SubCommand } from '#utils/types';
-
-import { version as azleVersion } from '../package.json';
+import {
+    Command as StableCommand,
```
[the remark this answers is not in the document]
Comment (bdemann) [reply src/build.ts:5]: done

File: src/stable/build/commands/new.ts [hunk 20]
```diff
@@ -19,13 +16,14 @@ export async function runCommand(
 
     const projectName = process.argv[3];
 
-    await copy(templatePath, projectName);
+    await cp(templatePath, projectName, { recursive: true });
 
-    const packageJson = (
+    const packageJsonPath = join(projectName, 'package.json');
```
[the remark this answers is not in the document]
Comment (bdemann) [reply src/stable/build/commands/new.ts:21]: done

File: src/stable/build/commands/new.ts [hunk 21]
```diff
@@ -34,13 +32,16 @@ export async function runCommand(
...
 
-    await outputFile(
-        join(projectName, 'package.json'),
+    await mkdir(dirname(packageJsonPath), { recursive: true });
+    await writeFile(
+        packageJsonPath,
         JSON.stringify(parsedPackageJson, null, 4)
     );
 
+    const tsConfigPath = join(projectName, 'tsconfig.json');
```
[the remark this answers is not in the document]
Comment (bdemann) [reply src/stable/build/commands/new.ts:41]: done

File: src/experimental/build/index.ts [hunk 22]
```diff
@@ -0,0 +1,269 @@
...
+        'examples',
+        httpServer === true ? 'experimental' : 'stable',
+        'demo',
+        projectName
+    );
+
+    await runStableNewCommand(azleVersion, templatePath, httpServer === true);
+}
+
+async function assertAzleExperimentalDeps(): Promise<void> {
```
[the remark this answers is not in the document]
Comment (bdemann) [reply src/experimental/build/index.ts:144]: done

File: src/experimental/build/index.ts [hunk 23]
```diff
@@ -0,0 +1,269 @@
...
+}
+
+async function getAzleExperimentalDepsVersion(
+    resolveDir: string
+): Promise<string | undefined> {
+    const packageJsonPath = join(resolveDir, 'package.json');
+    const packageJson = JSON.parse(await readFile(packageJsonPath, 'utf8'));
+    const packageName = 'azle-experimental-deps';
+    return (
+        packageJson.dependencies?.[packageName] ??
```
[the remark this answers is not in the document]
Comment (bdemann) [reply src/experimental/build/index.ts:184]: done

File: src/experimental/build/index.ts [hunk 24]
```diff
@@ -0,0 +1,269 @@
...
+import { runCommand as runStableDevTemplateCommand } from '#commands/dev/template';
+import { runCommand as runStableNewCommand } from '#commands/new';
+import { findProjectRoot } from '#experimental/build/utils/find_project_root';
+import { runCommand as runBuildCommand } from '#experimental/commands/build/index';
+import { runCommand as runDevTemplateCommand } from '#experimental/commands/dev/template';
+import { runCommand as runUploadAssetsCommand } from '#experimental/commands/upload_assets/index';
+import { getCanisterConfig } from '#utils/get_canister_config';
+import { AZLE_ROOT } from '#utils/global_paths';
+import {
+    Command as StableCommand,
```
[the remark this answers is not in the document]
Comment (bdemann) [reply src/experimental/build/index.ts:25]: done

File: src/experimental/build/index.ts [hunk 25]
```diff
@@ -0,0 +1,269 @@
...
+}
+
+async function handleDevTemplateCommand(ioType: IOType): Promise<void> {
+    const all = process.argv.includes('--all');
+
+    if (all === true) {
+        await runStableDevTemplateCommand(ioType);
+        await runDevTemplateCommand(ioType);
+    } else {
+        const experimental =
```
[the remark this answers is not in the document]
Comment (bdemann) [reply src/experimental/build/index.ts:99]: done

File: src/stable/test/deep_equal.ts (hunk 12 above)
Comment (bdemann) [reply src/stable/test/deep_equal.ts:13]: done

File: src/stable/test/deep_equal.ts (hunk 14 above)
Comment (bdemann) [reply src/stable/test/deep_equal.ts:41]: done

File: src/stable/test/deep_equal.ts (hunk 15 above)
Comment (bdemann) [reply src/stable/test/deep_equal.ts:62]: done

File: src/stable/test/deep_equal.ts (hunk 17 above)
Comment (bdemann) [reply src/stable/test/deep_equal.ts:78]: done

File: src/stable/test/deep_equal.ts (hunk 16 above)
Comment (bdemann) [reply src/stable/test/deep_equal.ts:68]: 1. Non-JSON types collapse or disappear
     - Anything that isn’t valid JSON (functions, undefined, Symbol(), BigInt, Map/Set, typed-arrays, custom class instances, etc.) will either be dropped or coerced into plain objects/strings/null by JSON.stringify.
     - e.g.
          - deepEqual({ a: undefined }, {}) → true (because both normalize to {})
          - deepEqual(new Set([1]), new Set([2])) → true (both normalize to {})
          - deepEqual(NaN, null) → true (because NaN → null)
2. Numeric precision & special numbers
     - JSON only supports finite IEEE‐754 doubles. Anything outside safe integer range loses precision.
     - NaN, Infinity, -Infinity all serialize to null, so you can’t distinguish them from an actual null.
3. Sparse arrays vs explicit null
     - Holes in arrays become null under JSON:
     `JSON.parse(JSON.stringify([, 42])) // => [null,42]`
     `deepEqual([,42], [null,42])        // => true`
If you actually care about distinguishing a hole from a real null, this won’t catch it.
4. Prototype & special object types
     - After parse, every object is just a plain {} or [].
     - You lose everything about Date, RegExp, custom prototypes, etc.
     - Those all get compared purely by their JSON shape (e.g. a Date becomes its ISO string).
5. Circular references
     - JSON.stringify will throw on cyclic data. You’ll never get to the deep-compare stage.
6. Ordering & performance
     - You do a nested keys2.includes(key) for every property, which is O(n²) if you have a very large object.
     - Key‐order doesn’t matter semantically (you only compare sets), but internally JSON’s order is driven by ES enumeration rules—if you ever switch to a sorter in jsonStringify, you could get different results.
7. Stack overflows on extreme nesting
     - It’s a purely recursive walk; extremely deep structures can blow the call stack.

Comment (bdemann) [reply src/stable/test/deep_equal.ts:68]: In designing this deepEquals function I took into account most of these. Keeping in mind that this function does not need to work on arbitrary values but rather will only work on json version of valid candid really limits how many weird edges cases we have. Our json parser easily handles things like NaNs, Infinity, sparce arrays, etc. And things like Sets that don't translate into json (and does give a false positive with this function) are not going to show up in candid values and if they did we would need to update our jsonStringify method. We don't have to worry about special object types like Date or RegExp because those are not supported in candid, and if they were we would need to update our jsonStringify.

The strength of this method really comes from our jsonStringify method. It ensures that all valid candid values can get properly translated into a json string which makes comparisons like this really rather simple. 

Ideally we could just compare the two strings, but to do that we would need to add something into the jsonStringify to make sure that there was a consistent ordering of properties, like alphabetical or something. Lacking that, I think that this solution is very robust and perfectly acceptable

File: .github/workflows/run_test.yml [hunk 26]
```diff
@@ -144,6 +144,13 @@ jobs:
                   command: 'npm install'
                   working-directory: ${{ matrix.test.path }}
 
+            - name: Install experimental dependencies for experimental tests
```
Comment (lastmjs) [inline .github/workflows/run_test.yml:147]: Explain in the name that this installation is only for the stable API tests running in experimental mode

File: .github/workflows/run_test.yml [hunk 27]
```diff
@@ -144,6 +144,13 @@ jobs:
...
+              if: matrix.azle_source == 'link' && ${{ inputs.run-experimental }}
+              uses: ./.github/actions/retry_command
+              with:
+                  command: 'npm install github:demergent-labs/azle-experimental-deps#34cf2e34958e03cae7bf63f77f86ae682eef8a7c'
```
Comment (lastmjs) [inline .github/workflows/run_test.yml:151]: Let's use the https URL in our npm install commands

File: examples/experimental/test/end_to_end/candid_rpc/async_await/package.json [hunk 28]
```diff
@@ -8,6 +8,6 @@
     },
     "dependencies": {
         "azle": "0.31.0",
-        "azle-experimental-deps": "github:demergent-labs/azle-experimental-deps"
+        "azle-experimental-deps": "github:demergent-labs/azle-experimental-deps#34cf2e34958e03cae7bf63f77f86ae682eef8a7c"
```
Comment (lastmjs) [inline examples/experimental/test/end_to_end/candid_rpc/async_await/package.json:11]: Check that installing with the https url doesn't change this, if it does let's change

File: src/stable/test/deep_equal.ts (hunk 16 above)
Comment (lastmjs) [reply src/stable/test/deep_equal.ts:68]: Rename the function

Comment (lastmjs) [reply src/stable/test/deep_equal.ts:68]: Rename `deepEqual` and `testEquality` to `candidDeepEqual` and `candidTestEquality`. In the JSDocs explain that these functions are only designed to be used with Candid values (which means values deserialized from serialized Candid).

File: src/stable/test/deep_equal.ts [hunk 29]
```diff
@@ -32,24 +27,36 @@ export function deepEqual(obj1: any, obj2: any): boolean {
...
+
     if (Array.isArray(obj1Json)) {
-        if (obj1Json.length !== obj2Json.length) {
+        return areEqualArrays(obj1Json, obj2Json);
+    }
+
+    return areEqualObjects(obj1Json, obj2Json);
+}
+
+function areEqualArrays(obj1Json: any, obj2Json: any): boolean {
```
Comment (lastmjs) [inline src/stable/test/deep_equal.ts:47]: add `any[]` or `Array<any>`

File: src/experimental/build/index.ts [hunk 30]
```diff
@@ -139,37 +116,39 @@ async function handleNewCommand(): Promise<void> {
...
 
-    if (ourAzleExperimentalDepsVersion === undefined) {
-        const beChill = true;
-        if (beChill === true) {
-            throw new Error('Unreachable');
-        }
+    if (theirAzleExperimentalDepsVersion === undefined) {
         throw new Error(
-            'Something has gone horribly wrong. Please open an issue right away, this must be fixed in azle immediately. Reach out to Jordan and tell him that Some How Azle Suffers Total Annihilation. He will know what to do.'
+            `azle-experimental-deps is not installed. Please run \`npm install azle-experimental-deps@${azleExperimentalDepsVersion}\``
```
Comment (lastmjs) [inline src/experimental/build/index.ts:145]: Needs to be the github https url

File: src/experimental/build/index.ts [hunk 31]
```diff
@@ -81,31 +71,18 @@ async function handleDevCommand(): Promise<boolean> {
...
-            process.env.AZLE_EXPERIMENTAL === 'true';
-
-        if (experimental === false) {
-            await runStableDevTemplateCommand(ioType);
-        } else {
-            await runDevTemplateCommand(ioType);
-        }
-    }
+async function handleDevTemplateCommand(ioType: IOType): Promise<boolean> {
+    const needsToInstallStable = process.argv.includes('--stable');
```
Comment (lastmjs) [inline src/experimental/build/index.ts:81]: --all

File: src/experimental/build/index.ts [hunk 32]
```diff
@@ -139,37 +116,39 @@ async function handleNewCommand(): Promise<void> {
...
+
+async function installAzleExperimentalDepsForNewProject(
+    projectName: string
+): Promise<void> {
+    const packageJsonPath = join(projectName, 'package.json');
+    const currentPackageJsonContents = await readFile(packageJsonPath, {
+        encoding: 'utf-8'
+    });
+    let packageJson = JSON.parse(currentPackageJsonContents);
+    packageJson.devDependencies['azle-experimental-deps'] =
```
Comment (lastmjs) [inline src/experimental/build/index.ts:130]: `packageJson.dependencies` instead of `devDependencies`

Comment (lastmjs) [reply src/experimental/build/index.ts:130]: This should be done everywhere I think

**Review by lastmjs — CHANGES_REQUESTED** [review]

File: src/stable/test/deep_equal.ts (hunk 16 above)
Comment (bdemann) [reply src/stable/test/deep_equal.ts:68]: done

File: .github/workflows/run_test.yml (hunk 26 above)
Comment (bdemann) [reply .github/workflows/run_test.yml:147]: done

File: examples/experimental/test/end_to_end/candid_rpc/async_await/package.json (hunk 28 above)
Comment (bdemann) [reply examples/experimental/test/end_to_end/candid_rpc/async_await/package.json:11]: it doesn't

File: src/stable/test/deep_equal.ts (hunk 29 above)
Comment (bdemann) [reply src/stable/test/deep_equal.ts:47]: done

File: src/experimental/build/index.ts (hunk 31 above)
Comment (bdemann) [reply src/experimental/build/index.ts:81]: done

File: src/experimental/build/index.ts (hunk 32 above)
Comment (bdemann) [reply src/experimental/build/index.ts:130]: done

File: src/experimental/build/index.ts (hunk 30 above)
Comment (bdemann) [reply src/experimental/build/index.ts]: done

File: .github/workflows/run_test.yml (hunk 27 above)
Comment (bdemann) [reply .github/workflows/run_test.yml]: done

**Comment by bdemann** [comment]
closes https://github.com/demergent-labs/azle/issues/2958

**Review by lastmjs — CHANGES_REQUESTED** [review]

File: src/stable/test/jest_link.ts (hunk 10 above)
Comment (lastmjs) [reply src/stable/test/jest_link.ts:3]: Please remove once these removal comments are the last things

File: examples/stable/test/property/candid_rpc/canister_methods/init/test/generate_tests.ts [hunk 33]
```diff
@@ -3,9 +3,8 @@ import { CandidValueAndMeta } from 'azle/experimental/_internal/test/property/ar
 import { CorrespondingJSType } from 'azle/experimental/_internal/test/property/arbitraries/candid/corresponding_js_type';
 import {
     AzleResult,
-    Test,
-    testEquality
-} from 'azle/experimental/_internal/test/property/test';
+    candidTestEquality,
+    Test} from 'azle/experimental/_internal/test/property/test';
```
[the remark this answers is not in the document]
Comment (bdemann) [reply examples/stable/test/property/candid_rpc/canister_methods/init/test/generate_tests.ts:7]: It looks like for some reason the formatter didn't run after a find and replace. So there were a lot of files that changed in what I thought was a trivial way and which should have been autoformatted and so unfortunately I didn't think to check them to make sure they were.

File: examples/stable/test/property/candid_rpc/canister_methods/inspect_message/test/generate_tests.ts [hunk 34]
```diff
@@ -5,9 +5,8 @@ import { CandidValueAndMeta } from 'azle/experimental/_internal/test/property/ar
 import { CorrespondingJSType } from 'azle/experimental/_internal/test/property/arbitraries/candid/corresponding_js_type';
 import {
     AzleResult,
-    Test,
-    testEquality
-} from 'azle/experimental/_internal/test/property/test';
+    candidTestEquality,
+    Test} from 'azle/experimental/_internal/test/property/test';
```
[the remark this answers is not in the document]
Comment (bdemann) [reply examples/stable/test/property/candid_rpc/canister_methods/inspect_message/test/generate_tests.ts:9]: yeah there are a lot of them unfortunately

File: examples/stable/test/property/candid_rpc/canister_methods/query/test/generate_tests.ts [hunk 35]
```diff
@@ -4,9 +4,8 @@ import { CandidValueAndMeta } from 'azle/experimental/_internal/test/property/ar
 import { CorrespondingJSType } from 'azle/experimental/_internal/test/property/arbitraries/candid/corresponding_js_type';
 import {
     AzleResult,
-    Test,
-    testEquality
-} from 'azle/experimental/_internal/test/property/test';
+    candidTestEquality,
+    Test} from 'azle/experimental/_internal/test/property/test';
```
[the remark this answers is not in the document]
Comment (bdemann) [reply examples/stable/test/property/candid_rpc/canister_methods/query/test/generate_tests.ts:8]: sure is

File: README.md (hunk 8 above)
Comment (bdemann) [reply README.md:37]: done

File: .github/workflows/run_test.yml [hunk 36]
```diff
@@ -144,11 +144,11 @@ jobs:
                   command: 'npm install'
                   working-directory: ${{ matrix.test.path }}
 
-            - name: Install experimental dependencies for experimental tests
+            - name: Install experimental dependencies for stable tests if running in experimental mode
```
[the remark this answers is not in the document]
Comment (bdemann) [reply .github/workflows/run_test.yml:147]: done

**Review by lastmjs — DISMISSED** [review]

File: .github/workflows/run_test.yml [hunk 37]
```diff
@@ -115,6 +115,13 @@ jobs:
                   command: 'npm install'
                   working-directory: ${{ matrix.test.path }}
 
+            - name: Install experimental dependencies for stable api tests if running in experimental mode
+              if: matrix.azle_source == 'link' && ${{ inputs.run-experimental }}
```
Comment (lastmjs) [inline .github/workflows/run_test.yml:119]: What happens if azle_source is npm though? Don't we need to install azle-experimental-deps then as well?

**Review by lastmjs — CHANGES_REQUESTED** [review]

**Review by lastmjs — APPROVED** [review]
